packages/contracts-bedrock/src/L2/DependencyManager.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Libraries
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { Unauthorized } from "src/libraries/errors/CommonErrors.sol";
+import { Constants } from "src/libraries/Constants.sol";
+
+// Interfaces
+import { ISemver } from "interfaces/universal/ISemver.sol";
+import { IL2ToL1MessagePasser } from "interfaces/L2/IL2ToL1MessagePasser.sol";
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+
+/// @custom:proxied true
+/// @custom:predeploy 0x4200000000000000000000000000000000000029
+/// @title DependencyManager
+/// @notice The DependencyManager contract is used to manage the interop dependency set. This set contains the chain IDs
+///         that the current chain is dependent on. When updating the dependency set, the DependencyManager will
+///         initiate a withdrawal tx to update the dependency set on L1.
+contract DependencyManager is ISemver {
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    /// @notice Error when the interop dependency set size is too large.
+    error DependencySetSizeTooLarge();
+
+    /// @notice Error when a chain ID already in the interop dependency set is attempted to be added.
+    error AlreadyDependency();
+
+    /// @notice Event emitted when a new dependency is added to the interop dependency set.
+    event DependencyAdded(uint256 indexed chainId, address indexed systemConfig, address indexed superchainConfig);
+
+    /// @notice The interop dependency set, containing the chain IDs in it.
+    EnumerableSet.UintSet internal _dependencySet;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @notice Adds a new dependency to the dependency set. This function is only callable by the derivation pipeline.
+    ///         It will initiate a withdrawal tx to update the dependency set on L1.
+    /// @param _superchainConfig    Address of the SuperchainConfig contract on L1.
+    /// @param _chainId             The new chain's ID to add to the dependency set.
+    /// @param _systemConfig        The new chain's SystemConfig contract address on L1.
+    function addDependency(address _superchainConfig, uint256 _chainId, address _systemConfig) external {
+        if (msg.sender != Constants.DEPOSITOR_ACCOUNT) revert Unauthorized();
+
+        if (_dependencySet.length() == type(uint8).max) revert DependencySetSizeTooLarge();
+
+        if (_chainId == block.chainid || !_dependencySet.add(_chainId)) revert AlreadyDependency();
+
+        // Initiate a withdrawal tx to update the dependency set on L1.
+        IL2ToL1MessagePasser(payable(Predeploys.L2_TO_L1_MESSAGE_PASSER)).initiateWithdrawal(
+            _superchainConfig,
+            400_000, // TODO: find an arbitrary value that is enough
```

Comment:
In the future, I want to make a new "type" of withdrawal message that doesn't have a gas limit for the call and there is a `require` on the success of the call. This would simplify so much. For now we have to deal with this whole gaslimit thing and be careful of the footgun